### PR TITLE
Js dom

### PR DESCRIPTION
**Pull request by Levi123**
### Js DOM

[Demo](https://levi123.github.io/dom-js/) |
[Code base](https://github.com/Levi123/dom-js/tree/feature/product)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by Levi123** [comment]
@A-Ostrovnyy Привіт, ви могли б глянути мій PR?

**Comment by A-Ostrovnyy** [comment]
@Levi123 As soon as I get to your PR, but for now don't waste time and focus on other tasks

**Comment by Levi123** [comment]
> @Levi123Як тільки я прийду до вашого піару, а поки не втрачайте час і зосередьтеся на інших завданнях

Thank you for answer

File: submissions/Levi123/domjs/script.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        if (value.id === buttonId){
+            mainTitle.innerHTML = `About BMW ${value.series}`;
+            mainImage.src = value.images;
+            mainText.innerHTML = value.information;
+        }
+    })
+}
+
+buttons.forEach((button) => {
+    button.addEventListener('click', generateInformationAboutModel);
```
Comment (madmaxWMFU) [inline submissions/Levi123/domjs/script.js:53]: Read about [event delegation](https://javascript.info/event-delegation)

File: submissions/Levi123/domjs/script.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+]
+
+
+const buttons = document.querySelectorAll('.button');
+let mainTitle = document.querySelector('.about-product__main-title');
+let mainImage = document.querySelector('.about-product__main-image');
+let mainText = document.querySelector('.about-product__description')
+
+function generateInformationAboutModel(event){
+    console.log('click');
```
Comment (madmaxWMFU) [inline submissions/Levi123/domjs/script.js:41]: Delete a debug code line.

File: submissions/Levi123/domjs/script.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    let buttonId = Number(event.target.id);
+    informationAboutModel.forEach((value) => {
+        if (value.id === buttonId){
+            mainTitle.innerHTML = `About BMW ${value.series}`;
+            mainImage.src = value.images;
+            mainText.innerHTML = value.information;
```
Comment (madmaxWMFU) [inline submissions/Levi123/domjs/script.js:47]: In this task, you must show work with DOM elements. Create elements or generate a wrap of content and render them.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
In this task, you don't created a new task branch. As a result, we have code from the previous task.

File: submissions/Levi123/domjs/script.js (hunk 1 above)
Comment (Levi123) [reply submissions/Levi123/domjs/script.js:53]: > Thank you! I must do it with event delegation? can i leave it as it is?

File: submissions/Levi123/domjs/index.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            <div class="about-product__info-block">
+                <h1 class="about-product__main-title">BMW 1 series</h1>
+                <img src="https://www.bmw.ua/content/dam/bmw/common/all-models/1-series/5-door/2019/inspire/bmw-1-series-inspire-taphold-driving-dynamics-desktop-01.jpg/jcr:content/renditions/cq5dam.resized.img.1185.large.time1627482147001.jpg" alt="bmw 1 series" class="about-product__main-image">
+                <p class="about-product__description">Чисте прагнення: новий BMW 1 серії прибув, щоб привернути до себе захоплені погляди. Його видатний дизайн і прогресивна мова дизайну виділяють його з натовпу. Те ж саме стосується і його салону із сучасною атмосферою, яка надихає своєю першокласною якістю та відкритістю. Ефективні двигуни, передній привід, вперше представлений в новій BMW 1 серії, і новітні технології шасі гарантують враження від керування, повні динаміки і маневреності, а інноваційні технології та системи.</p>
+            </div>
+        </div>
+    </section>
+    <script src="script.js"></script>
+</body>
+</html>
```
[the remark this answers is not in the document]
Comment (Levi123) [reply submissions/Levi123/domjs/index.html]: done

File: submissions/Levi123/domjs/script.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        information: 'BMW 5 серії Седан — втілення спортивного бізнес-седана. З першого погляду він демонструє стильний атлетизм, який особливо виражається у динамічному силуеті та лаконічному дизайні. Прості поверхні та чіткі контури надають елегантному зовнішньому вигляду сучасну та технологічну атмосферу. Елегантний і функціональний інтерєр доповнює інноваційні амбіції автомобіля технологіями майбутнього та спортивними особливостями. зкомпромісний нонконформіст: абсолютно новий BMW 4 серії Coupé рішуче та елегантно виходить за рамки всіх сучасних правил. Незалежний дизайн і класичний силует купе підкреслюють його прогресивний характер. Спортивна легкість плавно переходить на дорогу, де вражаюча маневреність і привабливі характеристики сприяють підвищенню рівня адреналіну.',
+    },
+]
+
+
+const buttons = document.querySelectorAll('.button');
+let mainTitle = document.querySelector('.about-product__main-title');
+let mainImage = document.querySelector('.about-product__main-image');
+let mainText = document.querySelector('.about-product__description')
```
[the remark this answers is not in the document]
Comment (Levi123) [reply submissions/Levi123/domjs/script.js]: done

File: submissions/Levi123/domjs/script.js (hunk 1 above)
Comment (Levi123) [reply submissions/Levi123/domjs/script.js]: done

File: submissions/Levi123/domjs/script.js (hunk 2 above)
Comment (Levi123) [reply submissions/Levi123/domjs/script.js]: done

File: submissions/Levi123/domjs/script.js (hunk 3 above)
Comment (Levi123) [reply submissions/Levi123/domjs/script.js:51]: done

File: submissions/Levi123/domjs/script.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+            mainInfoBlock.append(mainImage);
+            mainImage.classList.add('about-product__main-image');
+            mainImage.src = value.images;
+        }
+    })
+}
+
+button.addEventListener('click', function (event) {
+    if (event.target.closest('button')){
```
Comment (madmaxWMFU) [inline submissions/Levi123/domjs/script.js:61]: I think this was better `button.addEventListener('click', generateInformationAboutModel)`

File: submissions/Levi123/domjs/index.html [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+                <button
+                    href="#" 
+                    class="about-product_navigation-menu button" id="3">3 series
+                </button>
+                <button
+                    href="#" 
+                    class="about-product_navigation-menu button" id="4">4 series
+                </button>
+                <button
+                    href="#"
```
Comment (madmaxWMFU) [inline submissions/Levi123/domjs/index.html:46]: Did the tag `button` has the attribute `href`?

File: submissions/Levi123/domjs/index.html [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+                    class="about-product_navigation-menu button" id="3">3 series
+                </button>
+                <button
+                    href="#" 
+                    class="about-product_navigation-menu button" id="4">4 series
+                </button>
+                <button
+                    href="#" 
+                    class="about-product_navigation-menu button" id="5">5 series
+                </button>
```
Comment (madmaxWMFU) [inline submissions/Levi123/domjs/index.html:48]: It is better to use the dataset properties.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Update your local repository to see demo.

File: submissions/Levi123/domjs/script.js (hunk 6 above)
Comment (Levi123) [reply submissions/Levi123/domjs/script.js:61]: > Я думаю, що це було краще`button.addEventListener('click', generateInformationAboutModel)`

but I'm done with delegation? didn't need to do that?

File: submissions/Levi123/domjs/index.html (hunk 8 above)
Comment (Levi123) [reply submissions/Levi123/domjs/index.html:48]: > Краще використовувати властивості набору даних.

don't understand you

File: submissions/Levi123/domjs/index.html (hunk 7 above)
Comment (Levi123) [reply submissions/Levi123/domjs/index.html]: done

File: submissions/Levi123/domjs/script.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+
+const button = document.querySelector('.about-product__navigation')
+const mainInfoBlock = document.querySelector('.about-product__info-block');
+const mainTitle = document.createElement('h1');
+const mainImage = document.createElement('img');
+const mainText = document.createElement('p');
+
+function generateInformationAboutModel(event){
+    let buttonId = Number(event.target.id);
```
[the remark this answers is not in the document]
Comment (Levi123) [reply submissions/Levi123/domjs/script.js]: done

File: submissions/Levi123/domjs/index.html (hunk 8 above)
Comment (madmaxWMFU) [reply submissions/Levi123/domjs/index.html:43]: Read about [data attributes](https://developer.mozilla.org/ru/docs/Learn/HTML/Howto/Use_data_attributes)

File: submissions/Levi123/domjs/script.js (hunk 6 above)
Comment (madmaxWMFU) [reply submissions/Levi123/domjs/script.js:61]:
```javascript
const menuList = document.querySelector('.about-product__navigation');
buttons.addEventListener('click', generateInformationAboutModel);

```

Comment (Levi123) [reply submissions/Levi123/domjs/script.js:61]: > ```js
> const menuList = document.querySelector('.about-product__navigation');
> buttons.addEventListener('click', generateInformationAboutModel);
> ```

Sorry me, but i don't understand you

File: submissions/Levi123/domjs/index.html (hunk 8 above)
Comment (Levi123) [reply submissions/Levi123/domjs/index.html:43]: > Read about [data attributes](https://developer.mozilla.org/ru/docs/Learn/HTML/Howto/Use_data_attributes)

done

File: submissions/Levi123/domjs/script.js (hunk 6 above)
Comment (Levi123) [reply submissions/Levi123/domjs/script.js]: > ```js
> const menuList = document.querySelector('.about-product__navigation');
> buttons.addEventListener('click', generateInformationAboutModel);
> ```

I returned to the previous implementation method

File: submissions/Levi123/domjs/script.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+            mainInfoBlock.append(mainImage);
+            mainImage.classList.add('about-product__main-image');
+            mainImage.src = value.images;
+        }
+    })
+}
+
+buttons.forEach((button) => {
+    button.addEventListener('click', generateInformationAboutModel);
```
Comment (madmaxWMFU) [inline submissions/Levi123/domjs/script.js:61]: You must add an event listener only for the menu list.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Update your local demo so I can see it.

File: submissions/Levi123/domjs/script.js (hunk 10 above)
Comment (Levi123) [reply submissions/Levi123/domjs/script.js:61]: > You must add an event listener only for the menu list.

but I seem to have done so?

Comment (Levi123) [reply submissions/Levi123/domjs/script.js:61]: > You must add an event listener only for the menu list.

i have button class only on buttons and use it in this method

Comment (madmaxWMFU) [reply submissions/Levi123/domjs/script.js:61]: find me in telegram madmaxWMFU

Comment (A-Ostrovnyy) [reply submissions/Levi123/domjs/script.js:61]: [Read this article](https://davidwalsh.name/event-delegate#:~:text=Event%20delegation%20allows%20you%20to,just%20how%20event%20delegation%20works.) hope it helps)

**Review by madmaxWMFU — COMMENTED** [review]
Use [prettier](https://prettier.io/) to format the code. Sometimes you forget to put `;`.

**Review by madmaxWMFU — APPROVED** [review]
Great, I approve of your work. Good luck with future tasks!

**Comment by Levi123** [comment]
> Great, I approve of your work. Good luck with future tasks!

thanks for the help!

**Pull request merged**